### ISSUE TITLE
Color-code Rating val. to Custom Judgement Windows (in-game leaderboards)

### ISSUE THREAD

**Issue by Veeyu22** [Quaver/Quaver#3719]
### Color-code the Yellow Performance Rating Values on the Global Leaderboards to the Equivalent Template Judgement Window Modifier. (A real mouthful, I know)
(Standard - Impossible judgement window templates or custom to equivalent)

- **_WHY?_**
Because the player does not want to display/submit low accuracy/score and is then not incentivized to use them.
This is important because this is a huge part of Quaver to me and is criminally under-utilized or actively appreciated.
This gives incentive and introduces a bit more individuality & fun on the leaderboard "WTF this dude cleared this on Impossible?!"
This will also make people want to focus on acc more and in doing so, reach new heights.
Making this a larger deal opens the gates to a few other things too! (See "Extra Credit")
Also it looks neat and everyone likes showing off! B)
_See bottom/full for **simple** fixes to issues/disincentives (ie. Profile Accuracy)_

- _How is it used?_
**Colors used sync up with difficulty scaling colors in maps.** [Which works out very cleanly]
Standard - Default Yellow (same as map Diffs 10-19)
Strict - Red (Diffs 20-29)
Tough - Pink (Diffs 30-39)
Extreme - Purple (Diffs 40-49)
Impossible - White (Diffs 50+ off the charts!)

If possible confusion wants to be avoided, potentially include a [color coded?] (i) info circle that displays on-hover `"[##] is [Color] because they passed it with [Template Name] Judgements!"` ONCE on the top clear w/template per map
It's simple to identify once a player is aware, since we already know what the map colors mean.
(The info circle is the same as the one used to check accuracy needed to get a certain score if they are using increased speed, color coded change may be a good idea)

(Stating large fixes now before proceeding incase they were left hanging mentally)

- _How does it look?_ 
Like this? [mind the font]
![image](https://user-images.githubusercontent.com/42015732/209466045-427e22cf-8618-432f-88e5-b3a880bccce4.png)
Could look better if an outline is used instead of a full recolor?? Maybe color the combo and acc too??

- _Potential issues?_
**People still don't want to damage their profile acc scores**, either calculate for standard and add to total or just don't at all (easy fix? maybe this already happens?)
**_Custom_** judgements should **_snap to above colors by comparing_** ALL combined lowest values to ones used in the templates. NEEDS TO HAPPEN. Otherwise people will barely use the custom template feature, my favorite part!! 
Harder to implement than I initially thought, but holds good value in my opinion.
People use the feature too much and the leaderboard looks like a 5 set crayon box. (Yay?)
People with skills don't use it enough and it gets buried under regular clears. (Kind of doubt this, but we'll see)

- _Alternatives? Why not?_
I initially considered adding a form of mod icon but I think it lacks in a lot of areas. You could add the entire template name but that's really big, shorten it and it becomes unclear (S, T, E, I), color code the letters and it's a bit better, but I think as more mods become a thing it may end up too cluttered. You're essentially changing from a letter to a font color instead, and a single (i) on one clear per map to briefly explain it. Also, it looks cool, and the other one, less cool + harder to spot = less incentive for cool feature & important focus point.
What else? Clear borders/Backgrounds? I like it, but it might be too much color for the sleek grey look, and could look bad in contrast with other mods like speed mod on top of pink.
Ultimately, I chose to color the font and I think it's the right choice here.

- Extra Credit: 🖍️ 
Global Leaderboard sorting for highest scoring Impossible clears. (Supporter only?) This opens the floodgates to "Nobody cleared this on impossible yet"
Judgement related clear steam/profile achievements! 
90% acc or above: diff 20 on strict ["Strong Arm"], diff 30 on Tough ["Maximum Dexterity"], Extreme diff 40 ["Inhuman"], Impossible on Diff _40_ ["Ascended Being"]
*Maybe add some flavor to `"[Template Name] Judgements!"` via an extra exclamation mark to the end of "Judgements!" on Impossible top template clears if desired ;) Since we intend to only show the (i) once per map they would need to submit a score higher than all the other template users on the hardest template. It's silly but tiny secrets like these are fun.

- Closing comments:
Just pouring out a lot of love for Quaver, I'd really like to see this feature implemented and what people get up to with it, I think it helps show people the tools available to them to help improve and adds more accomplishments to reach to, achieve and be proud of.
If you can't see this being made to the light of day, that's alright. I'd make a branch if I had the skills to implement it myself. I just want to try to improve upon this game in the ways I'm currently able to.

_I apologize for a rather large reading_, but I wanted to help pave the vision and it's values out & weed out problems right away.
For that, you get a star ⭐ 
**If you scrolled through the whole thing and didn't read it**, you get nothing! Good day sir!
Bonus stars for sharing thoughts, criticism too. Be constructive, and I'll give you star 2. Everybody wants 🌟 
Thank you very much
-V

**Comment by Veeyu22** [comment]
Consider applying https://github.com/Quaver/Quaver/issues/3699 as well here, since this would be a general revitalization of the feature.

**Comment by IceDynamix** [comment]
> People still don't want to damage their profile acc scores, either calculate for standard and add to total or just don't at all (easy fix? maybe this already happens?)

> Because the player does not want to display/submit low accuracy/score and is then not incentivized to use them.

***all online scores are already automatically converted to standard accuracy on submission!!!*** any score on an online leaderboard or on a player's profile is shown in standard accuracy, only local scores keep their custom accuracy.

profile acc is based on your top plays (which are naturally high acc scores), so your profile acc doesn't get influenced by low acc scores because they will most likely not be in your top plays

> Colors used sync up with difficulty scaling colors in maps

adding colors to rating values that do not directly correlate to the actual difficulty value is confusing, it would be way more interesting to add the color to the accuracy instead 

> Custom judgements should snap to above colors by comparing ALL combined lowest values to ones used in the templates.

this is very ambiguous and finding a common denominator for all judgement windows is difficult, see for example a judgement window that looks like 18 / 36 / 76 / 76 / 76 / 76 where everything above a perfect is considered a miss

> Global Leaderboard sorting for highest scoring Impossible clears. (Supporter only?)

you can already see high accuracy skill being tested by using rate leaderboards, where only scores on the same rate as the currently selected one will be shown. most maps are too hard to X rank feasibly, so the competitive field for very high accuracy scores (99.95+) is broad enough to accommodate most players

> Judgement related clear steam/profile achievements!

hp passing is still a mess and achievements based on custom judgements are difficult to implement because the score needs to be verified server-side, but once the score is sent to the server it's already converted to standard accuracy as mentioned before.

---

this feature would require major infrastructure changes to accommodate custom judgements server-side, which is too out of scope to implement as of now.

it would also require adding additional skinning attributes and documentation on the wiki to explain the colors.

---

tl;dr too niche and too much effort to implement

**Comment by Veeyu22** [comment]
Beeeeg Gold star for you! 🌟

**_"All online scores are already automatically converted to standard on submission!!"_** 
Honestly maybe making this info more accessible in one form or another would be a good move, though I still feel it will remain heavily under used and under appreciated this way, but if it's niche then it may not be worth the hassle.
I just assumed it probably already existed because it would make sense to implement. 
And, if it is easily accessible, then I'm crazy for missing it! 

Also, very good point with the rate leaderboards, though this still doesn't solve the overall "Nobody uses templates on leaderboard or overall" issue, yes, technically speaking one who plays perfectly can just get X and call it a day but what you have to show for that journey requires clicking into the play and showing some small text on the right. At least as it currently exists the judgement window mod doesn't display in the scoring window easily. 

Too much effort? 
Yeah, honestly. It's listed under potential issues, I still think though that we have an underlying problem or two and there is value to resolving it in whole or in part. But I wasn't ready for what sounds WAY bigger than I initially expected, "major infrastructure changes" and the continued, very good points given after the fact. Which makes more sense, I should much more heavily consider scope in future suggestions and try keep them easy to implement. 

Overall, this is a very concise reply, I'll consider the arching issues I'd provided and see if there's an easier method (maybe the mod icon symbols are an easier alternative?) 

Closing for now because of scope issues. Thank you for the support IceDynamix, this was extremely helpful.
